lua/tabout/node.lua

```diff
@@ -33,16 +31,21 @@ end
 
 ---returns line,col or nil,nil
 ---@param dir string | "'forward'" | "'backward'"
----@return integer
----@return integer
+---@return TSNode?
 M.get_node_at_cursor = function(dir)
     local cursor = vim.api.nvim_win_get_cursor(0)
-    local cursor_range = {cursor[1] - 1, cursor[2]}
-    if dir == 'backward' and cursor[2] then
-        cursor_range = {cursor[1] - 1, cursor[2] - 1}
+    local line = cursor[1] - 1
+    local col = dir == 'backward' and cursor[2] and cursor[2] - 1
+    local cursor_range = {line, col, line, col}
+    local parser = vim.treesitter.get_parser(0)
+    if parser then
+      parser:parse()
+    else
+        logger.debug("get_node_at_cursor: No parser found for filetype " .. vim.bo[0].filetype)
+        return
     end
-    local root =
-        ts_utils.get_root_for_position(cursor_range[1], cursor_range[2])
+    local tree = parser:tree_for_range(cursor_range)
+    local root = tree and tree:root()
 
```

Comment:
Some changes here cause line 62 to throw an error for me on a simple go file:

```
//test.go
package main

func main(|) // hit tab here
```

Error about concatenating a bool value in line 62.

Can you double check or let me know what I am missing?

Reply:
Oops. Fixed :smile: